docker/api/exec_api.py

```diff
@@ -62,7 +63,13 @@ def exec_start(self, exec_id, detach=False, tty=False, stream=False):
             'Detach': detach
         }
 
+        # socket = True implies stream = True otherwise you'll get:
+        #   AttributeError: 'NoneType' object has no attribute '_sock'
+        stream = True if socket else stream
```

Comment:
This is a bit harder to read than it needs to be. How about just:

```python
if socket:
    stream = True
```

Also, this is just a personal preference, but I prefer to modify arguments at the very top of a function, so it's all in one place, so ideally this would go above the bit where we initialise `data`.